spacy_llm/tasks/summarization.py

```diff
@@ -0,0 +1,145 @@
+import warnings
+from typing import Any, Callable, Iterable, List, Optional, Type
+
+import jinja2
+from pydantic import BaseModel
+from spacy.language import Language
+from spacy.tokens import Doc
+from spacy.training import Example
+
+from ..registry import registry
+from ..ty import ExamplesConfigType
+from .templates import read_template
+from .util import SerializableTask
+from .util.serialization import ExampleType
+
+_DEFAULT_SUMMARIZATION_TEMPLATE_V1 = read_template("summarization")
+
+
+class SummarizationExample(BaseModel):
+    text: str
+    summary: str
+
+
+@registry.llm_tasks("spacy.Summarization.v1")
+def make_summarization_task(
+    template: str = _DEFAULT_SUMMARIZATION_TEMPLATE_V1,
+    examples: ExamplesConfigType = None,
+    max_n_words: Optional[int] = None,
+    field: str = "summary",
+):
+    """Summarization.v1 task factory.
+
+    template (str): Prompt template passed to the model.
+    examples (Optional[Callable[[], Iterable[Any]]]): Optional callable that
+        reads a file containing task examples for few-shot learning. If None is
+        passed, then zero-shot learning will be used.
+    max_n_words (int): Max. number of words to use in summary.
+    field (str): The name of the doc extension in which to store the summary.
+    """
+    raw_examples = examples() if callable(examples) else examples
+    span_examples = (
+        [SummarizationExample(**eg) for eg in raw_examples] if raw_examples else None
+    )
+
+    return SummarizationTask(
+        template=template, examples=span_examples, max_n_words=max_n_words, field=field
+    )
+
+
+class SummarizationTask(SerializableTask[SummarizationExample]):
+    def __init__(
+        self,
+        template: str = _DEFAULT_SUMMARIZATION_TEMPLATE_V1,
+        examples: Optional[List[SummarizationExample]] = None,
+        max_n_words: Optional[int] = None,
+        field: str = "summary",
+    ):
+        """Default summarization task.
+
+        template (str): Prompt template passed to the model.
+        examples (Optional[Callable[[], Iterable[Any]]]): Optional callable that
+            reads a file containing task examples for few-shot learning. If None is
+            passed, then zero-shot learning will be used.
+        max_n_words (int): Max. number of words to use in summary.
+        field (str): The name of the doc extension in which to store the summary.
+        """
+        self._template = template
+        self._examples = examples
+        self._max_n_words = max_n_words
+        self._field = field
+        self._prompt_examples = examples or []
+        self._are_example_summaries_checked = False
```

Comment:
This is a parameter that can't be changed? What is its goal? (and can we make it shorter :-))

Reply:
Ah maybe we should set it to `True` after the check happened?

Reply:
Yeah, forgot setting it to True after the check. Done in https://github.com/explosion/spacy-llm/pull/181/commits/e7b287a3ff35251b323d6e9566e2e5fcdb0f3977. Renamed in https://github.com/explosion/spacy-llm/pull/181/commits/b32a8165f981e558df8ec714f963e609a24a5e06.